tests/basic/extra_metrics_test.py

```diff
@@ -0,0 +1,31 @@
+from textwrap import dedent
+
+from tests.helpers.agent import Agent
+from tests.helpers.metadata import Metadata
+from tests.helpers.verify import verify
+from tests.monitors.expvar.expvar_test import run_expvar
+
+
+def test_extra_metrics_passthrough():
+    """
+    The specified extraMetrics should be allowed through even though they are
+    not included by default.
+    """
+    metadata = Metadata.from_package("expvar")
+
+    with run_expvar() as expvar_container_ip:
```

Comment:
btw you can have multiple context managers with one with statement that gets rid of multiple levels of indenting. Something like:

```python
with run_expvar() as expvar_container_ip, Agent.run(..):
   pass
```

Reply:
That would work if the agent config didn't depend on the container ip.

Reply:
It still works in that case:

```python
    with run_expvar() as expvar_container_ip, Agent.run(
        f"""
        monitors:
          - type: expvar
            host: {expvar_container_ip}
            port: 8080
            intervalSeconds: 1
            extraMetrics:
             - memstats.by_size.mallocs
        """
    ) as agent:
        assert "memstats.by_size.mallocs" in metadata.nonincluded_metrics
        verify(agent, metadata.included_metrics | {"memstats.by_size.mallocs"})
```

Reply:
It doesn't really bother me either way, just an fyi thing.